### PR TITLE
Feature/deploy ad campaign contract

### PR DESCRIPTION
**Pull request by achillegley**
# Description

app/src/components/forms/CreateAddCamapaign.vue
this form help define attributes and deploy adManagerCampaign.sol contract
image
test file: app/src/components/forms/__tests__/CreateAddCampaign.spec.ts

app/src/components/TeamContracts.vue
help list deployed adCampaignContracts. 
image
test file: app/src/components/__tests__/TeamContracts.spec.ts

app/src/components/TeamContractAdmins.vue
help list (fetch from deploed contrac) defined admins(managers ) for each contract.
image
test file : app/src/components/__tests__/TeamContractAdmins.spec.ts
action on click add admin : add new admin 
action on click remove admin: remove admin from existings

app/src/components/TeamContractsDetail.vue
help view deployed contract detail
image
test file: app/src/components/__tests__/TeamContractDetail.spec.ts

app/src/components/TeamContractEventList.vue
help retreive events for deployed contract
image

app/src/views/SingleTeamView.vue
new column "TeamContract " added to tabs (setabs)
image

Make sur you run the following commands before submitting your PR:

- /app

```bash
npm run build
npm run test:unit
npm run type-check
npm run lint
npm run format
```

- /backend

```bash
npm run build
npm run test
npm run lint
npm run format
```

- /contract

```bash
npm run test
npm run lint
npm run format
```

- **Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by hermannleboss** [comment]
@achillegley you do not generate migration, so it's create a 500 error on the API

**Comment by achillegley** [comment]
> @achillegley you do not generate migration, so it's create a 500 error on the API

@hermannleboss  should be ok now. i just  add it. thanks

File: app/src/components/TeamContractEventList.vue [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+</script>
+
+<style scoped>
+.campaign-list {
+  padding: 10px;
+}
+
+.campaign-item {
+  margin-bottom: 10px;
+}
```
Comment (hermannleboss) [inline app/src/components/TeamContractEventList.vue:88]: You can simply use tailwind for this

Comment (hermannleboss) [reply app/src/components/TeamContractEventList.vue:88]: The class you created for the css you should remove thems

File: app/src/components/TeamContracts.vue [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        </tr>
+      </thead>
+      <tbody>
+        <!-- row 1 -->
+        <tr v-for="(contract, index) in _contracts" :key="index" class="bg-base-200">
+          <th>{{ index + 1 }}</th>
+          <td>{{ contract.type }}</td>
+          <td><AddressToolTip :address="contract.address" class="text-xs" /></td>
+          <td>
+            <button @click="openAdminsModal(contract)" class="btn btn-ghost btn-xs">
```
Comment (hermannleboss) [inline app/src/components/TeamContracts.vue:23]:
```suggestion
            <button @click="openAdminsModal(contract)" class="btn btn-ghost btn-xs" data-test="open-admin-modal-btn">
```


by adding a data test attribute like that you can simply select the component you want for the test

File: app/src/components/__tests__/TeamContracts.spec.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+      props: { contracts, teamId: 'team1' },
+      global: {
+        plugins: [createPinia()]
+      }
+    })
+
+    // Simulate clicking the first row
+    const firstRow = wrapper.find('tbody tr')
+    const firstRowCells = firstRow.findAll('td')
+    await firstRowCells[4].trigger('click')
```
Comment (hermannleboss) [inline app/src/components/__tests__/TeamContracts.spec.ts:75]: You can simply use data test attribute as selector here, so you won't have error when the component place or position change in the future.

**Review by hermannleboss — COMMENTED** [review]
LGTM, you just need to apply the comments.

**Pull request merged**